lib/galaxy/tools/loader_directory.py

```diff
@@ -156,11 +171,19 @@ def looks_like_a_tool_cwl(path):
         return False
 
     file_class = as_dict.get("class", None)
+    if classes is None or file_class not in classes:
+        return False
```

Comment:
Is there a use-case for calling `looks_like_a_cwl_artifact(path, None)` ?

`is_a_yaml_with_class()` has no default value for `classes`.